src/binding/petsc4py/src/petsc4py/typing.py

```diff
@@ -89,18 +96,119 @@
 ArrayScalar = NDArray[Scalar]
 """Array of `Scalar` numbers."""
 
+AccessModeSpec = Literal['rw', 'r', 'w'] | None
+"""Access mode specification.
+
+   Possible values are:
+     - ``'rw'`` Read-Write mode.
+     - ``'r'`` Read-only mode.
+     - ``'w'`` Write-only mode.
+     - `None` as ``'rw'``.
+
+"""
+
+InsertModeSpec = InsertMode | bool | None
+"""Insertion mode specification.
+
+   Possible values are:
+     - `InsertMode.ADD_VALUES` Add new value to existing one.
+     - `InsertMode.INSERT_VALUES` Replace existing entry with new value.
+     - `None` as `InsertMode.INSERT_VALUES`.
+     - `False` as `InsertMode.INSERT_VALUES`.
+     - `True` as `InsertMode.ADD_VALUES`.
+
+"""
+
+LayoutSizeSpec = int | tuple[int, int]
+"""`int` or 2-`tuple` of `int` describing the layout sizes.
+
+   A single `int` indicates global size.
+   A `tuple` of `int` indicates ``(localsize, globalsize)``.
```

Comment:
Isn't this slightly easier to read?

```suggestion
   A `tuple` of `int` indicates ``(local_size, global_size)``.
```